test/functional/ui/highlight_spec.lua

```diff
@@ -303,3 +303,60 @@ describe('Default highlight groups', function()
     ]], {[1] = {foreground = Screen.colors.Red, background = Screen.colors.Green}})
   end)
 end)
+
+describe('New special highlight color attribute', function()
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new(25,10)
+    screen:attach()
+    screen:set_default_attr_ignore({
+      {bold=true, foreground=Screen.colors.Blue}
+    })
+  end)
+
+  it('used for undercurls', function()
```

Comment:
The issue was that Screen wasn't checking the 'special' attributes when testing for attribute equality. I rebased off master and pushed the fixed version. Had to modify one other test to account for the change.